store/v2/multi/migration.go

```diff
@@ -0,0 +1,64 @@
+package root
+
+import (
+	prefixdb "github.com/cosmos/cosmos-sdk/db/prefix"
+	"github.com/cosmos/cosmos-sdk/store/iavl"
+	"github.com/cosmos/cosmos-sdk/store/mem"
+	v1Store "github.com/cosmos/cosmos-sdk/store/rootmulti"
+	"github.com/cosmos/cosmos-sdk/store/transient"
+	"github.com/cosmos/cosmos-sdk/store/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// MigrationFromIAVLStoreToSMTStore will migrate the complete state from iavl to smt
+func MigrationFromIAVLStoreToSMTStore(rs *v1Store.Store, rootStore *Store) error {
+	// Collect stores to snapshot (only IAVL stores are supported)
+	type namedStore struct {
+		*iavl.Store
+		name string
+	}
+	var stores []namedStore
+	for key := range rs.GetStores() {
+		switch store := rs.GetCommitKVStore(key).(type) {
+		case *iavl.Store:
+			stores = append(stores, namedStore{name: key.Name(), Store: store})
+		case *transient.Store, *mem.Store:
+			continue
+		default:
+			continue
+		}
+	}
+
+	// make new smt store schema
+	if len(rootStore.schema) != 0 {
```

Comment:
We should move this check to the top so that it short-circuits if a schema already exists and avoids doing the rs.GetStores() iteration.